stdmacs.h

```diff
@@ -44,11 +44,6 @@
 
 #define NIL(p)	((p*)NULL)
 
-/* Cygwin defines _STDDEF_H with only one leading '_' */
```

Comment:
Running build from master branch on this repo succeeds and with this pr it fails
```
unix/arlib.c: In function ‘_cache_member’:
unix/arlib.c:521:41: warning: implicit declaration of function ‘offsetof’ [-Wimplicit-function-declaration]
  521 |       mp = (MEMPTR) malloc(sizeof(char)*offsetof(MEM,m_name[strlen(name)+1]));
      |                                         ^~~~~~~~
unix/arlib.c:43:1: note: ‘offsetof’ is defined in header ‘<stddef.h>’; did you forget to ‘#include <stddef.h>’?
   42 | #include "sysintf.h"
  +++ |+#include <stddef.h>
   43 | 
unix/arlib.c:521:50: error: expected expression before ‘MEM’
  521 |       mp = (MEMPTR) malloc(sizeof(char)*offsetof(MEM,m_name[strlen(name)+1]));
      |                                                  ^~~
make[2]: *** [Makefile:640: unix/arlib.o] Error 1
make[2]: Leaving directory '/home/micwoj92/pkg/dmake'
make[1]: *** [Makefile:662: all-recursive] Error 1
make[1]: Leaving directory '/home/micwoj92/pkg/dmake'
make: *** [Makefile:428: all] Error 2
```

fix is to either revert these line deletions or add `#include <stddef.h>` to unix/arlib.c.
I only tested this on Linux, gcc 13.2.1.